File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt
package com.sourcegraph.cody.agent.protocol

import com.sourcegraph.cody.agent.protocol_generated.Position

enum class AutocompleteTriggerKind(val value: String) {
  AUTOMATIC("Automatic"),
  INVOKE("Invoke"),
}

data class AutocompleteParams(
    val uri: String,
    val position: Position,
    val triggerKind: String? = AutocompleteTriggerKind.AUTOMATIC.value,
    val selectedCompletionInfo: SelectedCompletionInfo? = null
)

File: agent/bindings/csharp/AutocompleteParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class AutocompleteParams
  {
    [JsonProperty(PropertyName = "uri")]
    public string Uri { get; set; }
    [JsonProperty(PropertyName = "filePath")]
    public string FilePath { get; set; }
    [JsonProperty(PropertyName = "position")]
    public Position Position { get; set; }
    [JsonProperty(PropertyName = "triggerKind")]
    public TriggerKindEnum TriggerKind { get; set; } // Oneof: Automatic, Invoke
    [JsonProperty(PropertyName = "selectedCompletionInfo")]
    public SelectedCompletionInfo SelectedCompletionInfo { get; set; }

    public enum TriggerKindEnum
    {
      [EnumMember(Value = "Automatic")]
      Automatic,
      [EnumMember(Value = "Invoke")]
      Invoke,
    }
  }
}
